doc/stdenv/platform-notes.chapter.md

```diff
@@ -59,79 +57,59 @@ Some common issues when packaging software for Darwin:
     '';
   }
   ```
-  The package `xcbuild` can be used to build projects that really depend on Xcode. However, this replacement is not 100% compatible with Xcode and can occasionally cause issues.
 
-  Note: Some packages may hardcode an absolute path to `xcrun`, `xcodebuild`, or `xcode-select`. Those paths should be removed or replaced.
+#### Package requires `xcodebuild`
+
+The xcbuild package provides an `xcodebuild` command for packages that really depend on Xcode. This replacement is not 100% compatible and may run into some issues, but it is able to build many packages. To use `xcodebuild`, add `xcbuildHook` to your package’s `nativeBuildInputs`. It will provide a `buildPhase` for your derivation. You can use `xcbuildFlags` to specify flags to `xcodebuild` such as the required schema. If a schema has spaces in its name, you must set `__structuredAttrs` to `true`. See MoltenVK for an example of setting up xcbuild.
```

Comment:
We should also mention absolute paths to `/usr/bin/xcodebuild` and `/usr/libexec/PlistBuddy`, as they’re quite frequent.